src/net.h

```diff
@@ -424,8 +435,15 @@ class CConnman
     // Attack example: scraping addrs in real-time may allow an attacker
     // to infer new connections of the victim by detecting new records
     // with fresh timestamps (per self-announcement).
-    std::vector<CAddress> m_addrs_response_cache;
-    std::chrono::microseconds m_update_addr_response{0};
+    struct CachedAddrResponse {
+        std::vector<CAddress> m_addrs_response_cache;
+        std::chrono::microseconds m_update_addr_response{0};
+    };
+
+    // Keep a separate cache for other net (onion etc.) peers so that
```

Comment:
Why not one for every network? 1000 addresses per network is still only a tiny amount of memory.